### PR TITLE
Retain generics when using actor macro

### LINKED ISSUE #121: Derive receive for actors with generic arguments.
Hi! I have an Actor that is generic over a type and when using the macro `#[actor(SomeMsg)]` the compiler complains with `wrong number of type arguments: expected 1, found 0    expected 1 type argument`.
```rust
#[actor(SomeMsg)]
struct MyActor<T> {
    foo: T,
}
// ... generates
impl Receive<MyActorMsg> for MyActor { ... }
```
Doesn't seem like a big issue to fix but I haven't worked with procedural macros, I might try later but if anyone knows how to feel free to jump in :)

### PR DESCRIPTION
**Pull request by hardliner66**
Fixes #121 
Added a generic struct to riker-macros test
Additionally I added riker as a dev dependency to riker-macros, in order to re-use the types inside the tests instead of duplicating type definitions

### CODE REVIEW

File: riker-macros/tests/macro.rs [hunk 1]
```diff
@@ -38,77 +32,65 @@ impl Receive<String> for NewActor {
         println!("String");
     }
 }
-struct BasicActorRef;
-type Context<T> = Option<T>;
-
-trait Actor: Send + 'static {
```
Comment (igalic) [inline riker-macros/tests/macro.rs:44]: I'm confused
was this a test?
code duplication?
it's all gone now

Comment (hardliner66) [reply riker-macros/tests/macro.rs:44]: In order to test that the macro generates the correct code, the types used by the generated code must be in scope. In the old version, the code was duplicated. In the new version I added riker as a dev-dependecy so it that we can utilize the types directly inside the tests.

This reduces duplication and if the types in riker change but the macro isn't adapted to the change, the tests should fail and not silently keep working.

Comment (igalic) [reply riker-macros/tests/macro.rs:44]: i understood this the first explanation 😅but thank you for the summary 💜

**Comment by leenozara** [comment]
Very nice! I can't recall why I chose to duplicate the types instead of using the dev dependency.

@hardliner66 btw looks like you need to `pre-commit`.

**Comment by hardliner66** [comment]
I installed pre-commit and ran rustfmt. Should be good now.

**Pull request merged**